### PR TITLE
[DPE-5530] Hardening Userguide

### PR DESCRIPTION
**Pull request by deusebio**
First draft of the Hardening Guide for K8s, similar to what has been done in https://github.com/canonical/kafka-operator/pull/249 and https://github.com/canonical/spark-k8s-bundle/pull/51

@Batalex @marcoppenheimer @zmraul

### CODE REVIEW

File: docs/explanation/e-hardening.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+## Applications
+
+In the following we provide guidance on how to harden your deployment using:
+
+1. Base Images
+2. Kafka and ZooKeeper Security Upgrades
+3. Encryption 
+4. Authentication
+5. Monitoring and Auditing
```
Comment (izmalk) [inline docs/explanation/e-hardening.md:68]:
```suggestion
1. Base images
2. Charmed operator security upgrades
3. Encryption 
4. Authentication
5. Monitoring and auditing
```

File: docs/explanation/e-hardening.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+### Base images
+
+Charmed Kafka K8s and Charmed ZooKeeper K8s run on top of rockcraft-based image shipping the Apache Kafka and Apache ZooKeeper 
+distribution binaries built by Canonical, and available in the [Kafka release page](https://launchpad.net/kafka-releases) and 
+[ZooKeeper release page](https://launchpad.net/zookeeper-releases), respectively. Both images are based on Ubuntu 22.04. 
+The images that can be found in the [Charmed Kafka](https://github.com/canonical/charmed-spark-rock) and
```
Comment (izmalk) [inline docs/explanation/e-hardening.md:75]:
```suggestion
The images that can be found in the [Charmed Kafka rock](https://github.com/canonical/charmed-spark-rock) and 
```

File: docs/explanation/e-hardening.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+4. Authentication
+5. Monitoring and Auditing
+
+### Base images
+
+Charmed Kafka K8s and Charmed ZooKeeper K8s run on top of rockcraft-based image shipping the Apache Kafka and Apache ZooKeeper 
+distribution binaries built by Canonical, and available in the [Kafka release page](https://launchpad.net/kafka-releases) and 
+[ZooKeeper release page](https://launchpad.net/zookeeper-releases), respectively. Both images are based on Ubuntu 22.04. 
+The images that can be found in the [Charmed Kafka](https://github.com/canonical/charmed-spark-rock) and 
+[Charmed ZooKeeper](https://github.com/canonical/charmed-zookeeper-rock) Github repositories are used as the base
```
Comment (izmalk) [inline docs/explanation/e-hardening.md:76]:
```suggestion
[Charmed ZooKeeper rock](https://github.com/canonical/charmed-zookeeper-rock) Github repositories are used as the base 
```

File: docs/explanation/e-hardening.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+Charmed Kafka supports the following authentication layers:
+
+1. [SCRAM-based SASL Authentication](/t/charmed-kafka-k8s-how-to-manage-app/10293)
+2. [certificate-base Authentication (mTLS)](/t/create-mtls-client-credentials/11079)
+3. OAuth Authentication using [Hydra](/t/how-to-connect-to-kafka-using-hydra-as-oidc-provider/14610) or [Google](/t/how-to-connect-to-kafka-using-google-as-oidc-provider/14611)
+
+Each combination of authentication scheme and encryption is associated to the dedicated listener and it maps to a well-defined port. 
+Please refer to the [listener reference documentation](/t/charmed-kafka-k8s-documentation-reference-listeners/13270) for more information. 
+
+### Monitoring and Auditing
```
Comment (izmalk) [inline docs/explanation/e-hardening.md:115]:
```suggestion
### Monitoring and auditing
```

File: docs/explanation/e-hardening.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+External user access to Kafka is logged to the `kafka-authorizer.log` that is pushes to [Loki endpoint](https://charmhub.io/loki-k8s) and exposed via [Grafana](https://charmhub.io/grafana), both components being part of the COS stack.
+Access denials are logged at INFO level, whereas allowed accesses are logged at DEBUG level. Depending on the auditing needs, 
+customize the logging level either for all logs via the [`log_level`](https://charmhub.io/kafka-k8s/configurations?channel=3/stable#log_level) config option or 
+only tune the logging level of the `authorizerAppender` in the `log4j.properties` file. Refer to the Reference documentation, for more information about 
+the [file system paths](/t/charmed-kafka-k8s-documentation-reference-file-system-paths/13269).
+
+## Additional Resources
+
+For further information and details on the security and cryptographic specifications used by Charmed Kafka, please refer to the [Security Explanation page](/t/charmed-kafka-k8s-documentation-explanation-security/15715).
```
Comment (izmalk) [inline docs/explanation/e-hardening.md:134]:
```suggestion
For more information and details on the security and cryptography used by Charmed Kafka, see the [Security Explanation page](/t/charmed-kafka-k8s-documentation-explanation-security/15715).
```

**Review by izmalk — APPROVED** [review]
Looks good. I've added some small suggestions.

**Comment by deusebio** [comment]
Published documentation to [Discourse](https://discourse.charmhub.io/t/charmed-kafka-k8s-documentation-explanation-hardening-guide/15857)

**Pull request closed without merging**